### PR TITLE
add sliceable-dict

### PR DESCRIPTION
**Pull request by bernt-matthias**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [ ] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [ ] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by bernt-matthias** [comment]
@conda-forge/help-python, ready for review!`

File: recipes/sliceable-dict/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+  home: https://github.com/
+  summary: A simple dictionary that support slicing and multiple key selection.
+  description: |
+    The provided class SliceDict is an extended dictionary. It behaves exactly
+    as the buildin dict for single (hashable) keys, but adds some additional
+    features. Namely, getting and setting multiple keys at once, slicing with
+    integers and boolean selection.
+  license: BSD-3-Clause
+  license_family: GPL
+  license_file: LICENSE
```
Comment (chrisburr) [inline recipes/sliceable-dict/meta.yaml:39]: These are inconsistent

File: recipes/sliceable-dict/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+  run:
+    - python >=3.7
+    - typing_extensions
+
+test:
+  imports:
+    - sliceable_dict
+
+about:
+  home: https://github.com/
```
Comment (chrisburr) [inline recipes/sliceable-dict/meta.yaml:30]: This should be changed to an page for the package rather than the entirity of GitHub.

**Review by chrisburr — CHANGES_REQUESTED** [review]

File: recipes/sliceable-dict/meta.yaml (hunk 2 above)
Comment (bernt-matthias) [reply recipes/sliceable-dict/meta.yaml:30]:
```suggestion
  home: https://github.com/palmb/slice-dict
```

File: recipes/sliceable-dict/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+  summary: A simple dictionary that support slicing and multiple key selection.
+  description: |
+    The provided class SliceDict is an extended dictionary. It behaves exactly
+    as the buildin dict for single (hashable) keys, but adds some additional
+    features. Namely, getting and setting multiple keys at once, slicing with
+    integers and boolean selection.
+  license: BSD-3-Clause
```
Comment (bernt-matthias) [inline recipes/sliceable-dict/meta.yaml:37]:
```suggestion
  license: GPL-3.0 license
```

File: recipes/sliceable-dict/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+
+about:
+  home: https://github.com/palmb/slice-dict
+  summary: A simple dictionary that support slicing and multiple key selection.
+  description: |
+    The provided class SliceDict is an extended dictionary. It behaves exactly
+    as the buildin dict for single (hashable) keys, but adds some additional
+    features. Namely, getting and setting multiple keys at once, slicing with
+    integers and boolean selection.
+  license: GPL-3.0 license
```
Comment (bernt-matthias) [inline recipes/sliceable-dict/meta.yaml:37]:
```suggestion
  license: GPL-3.0
```

File: recipes/sliceable-dict/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+
+about:
+  home: https://github.com/palmb/slice-dict
+  summary: A simple dictionary that support slicing and multiple key selection.
+  description: |
+    The provided class SliceDict is an extended dictionary. It behaves exactly
+    as the buildin dict for single (hashable) keys, but adds some additional
+    features. Namely, getting and setting multiple keys at once, slicing with
+    integers and boolean selection.
+  license: GPL-3.0
```
Comment (bernt-matthias) [inline recipes/sliceable-dict/meta.yaml:37]:
```suggestion
  license: GPL-3.0-only
```

**Comment by bernt-matthias** [comment]
Thanks a lot for the comments @chrisburr . I hope I adressed all of them.

**Review by bgruening — APPROVED** [review]

**Comment by bgruening** [comment]
Thanks @bernt-matthias!

**Pull request merged**